heft-plugins/heft-lint-plugin/src/SarifFormatter.ts

```diff
@@ -0,0 +1,270 @@
+ *                  such as file paths, messages, and suppression details.
+ * @param options - An object (`ISerifFormatterOptions`) containing options for formatting:
+ *                  - `ignoreSuppressed`: Boolean flag to decide whether to ignore suppressed messages.
+ *                  - `eslintVersion`: Optional string to include the version of ESLint in the SARIF log.
+ * @returns The SARIF log (`ISarifLog`) containing information about the linting results in SARIF format.
+ */
+export function formatAsSARIF(results: IExtendedLintResult[], options: ISerifFormatterOptions): ISarifLog {
+  const { ignoreSuppressed, eslintVersion } = options;
+  const sarifRun: ISarifRun = {
+    tool: {
+      driver: {
+        name: 'ESLint',
+        informationUri: 'https://eslint.org',
+        rules: []
+      }
+    }
+  };
+
+  const sarifLog: ISarifLog = {
+    version: '2.1.0',
+    $schema: 'http://json.schemastore.org/sarif-2.1.0-rtm.5',
+    runs: [sarifRun]
+  };
+
+  if (typeof eslintVersion !== undefined) {
+    sarifRun.tool.driver.version = eslintVersion;
+  }
+  let executionSuccessful: boolean = true;
+
+  for (const result of results) {
+    const { filePath } = result;
+    let sarifFile: ISarifFile | undefined = sarifFiles.get(filePath);
+    if (sarifFile === undefined) {
+      const artifactIndex: number = sarifFiles.size;
+      const fileUrl: string = url.pathToFileURL(filePath).toString();
+
+      sarifFile = {
+        location: {
+          uri: fileUrl
+        }
+      };
+      sarifFiles.set(filePath, sarifFile);
+
+      const containsSuppressedMessages: boolean =
+        result.suppressedMessages && result.suppressedMessages.length > 0;
+      const messages: IMessage[] =
+        containsSuppressedMessages && !ignoreSuppressed
+          ? [...result.messages, ...result.suppressedMessages]
+          : result.messages;
+
+      if (messages.length > 0) {
+        for (const message of messages) {
+          const level: string = message.fatal || message.severity === 2 ? 'error' : 'warning';
+          const sarifRepresentation: ISarifRepresentation = {
+            level,
+            message: {
+              text: message.message
+            },
+            locations: [
+              {
+                physicalLocation: {
+                  artifactLocation: {
+                    uri: fileUrl,
+                    index: artifactIndex
+                  }
+                }
+              }
+            ]
+          };
+
+          if (message.ruleId) {
+            sarifRepresentation.ruleId = message.ruleId;
+
+            if (containsSuppressedMessages && !ignoreSuppressed) {
+              sarifRepresentation.suppressions = message.suppressions
+                ? message.suppressions.map((suppression: ISuppressedAnalysis) => {
+                    return {
+                      kind: suppression.kind === 'directive' ? 'inSource' : 'external',
```

Comment:
Same question for this. Is there a constant that you can compare against?

Reply:
I don't think a const exists that suppression.kind can be compared against. The @types/eslint has suppression.kind listed as strings.